### ISSUE TITLE
ParameterTypeHint: False positive when using PHP 8.5 closure in constant expression

### ISSUE THREAD

**Issue by stlrnz** [slevomat/coding-standard#1852]
Consider the following code implementing the PHP 8.5 _closure in constant expression_ feature in combination with an attribute.

```php
<?php

declare(strict_types=1);

namespace Demo;

use Attribute;
use Closure;

#[Attribute]
class DemoAttribute
{
    public function __construct(Closure $run)
    {
    }
}

class DemoClass
{
    public function __construct(
        #[DemoAttribute(run: static function (bool $closureParam): bool {
            return $closureParam;
        })]
        bool $param
    )
    {
    }
}
```

The `SlevomatCodingStandard.TypeHints.ParameterTypeHint` falsely detects `$closureParam` as method parameter and reports a missing type hint.
```text
 20 | ERROR | Method \Demo\DemoClass::__construct() does not have parameter
    |       | type hint nor @param annotation for its parameter $closureParam.
    |       | (SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingAnyTypeHint)
```
